app/models/container_template.rb

```diff
@@ -12,4 +12,44 @@ class ContainerTemplate < ApplicationRecord
   serialize :objects, Array
 
   acts_as_miq_taggable
+
+  def instantiate(params, project = nil)
+    ems = ext_management_system
+    project = container_project.name unless project
+    openshift_client = ems.connect
+    kubeclient = ems.connect(:service => 'kubernetes')
+
+    temp = {
+      :metadata   => {
+        :name      => name,
+        :namespace => project
+      },
+      :objects    => objects,
+      :parameters => params
+    }
+
+    processed_template = process(openshift_client, temp)
+    create_objects(openshift_client, kubeclient, processed_template['objects'], project) if processed_template
+  end
+
+  def process(client, temp)
+    client.process_template(temp)
+  rescue KubeException => e
+    $log.error("Unexpected Exception while processing template: #{e}")
+    false
+  end
+
+  def create_objects(openshift_client, kubeclient, objects, project)
+    objects.each do |obj|
+      obj = obj.symbolize_keys
+      obj[:metadata][:namespace] = project
+      create_entity = "create_#{obj[:kind].underscore}"
+      client = openshift_client.respond_to?(create_entity) ? openshift_client : kubeclient
```

Comment:
if you are creating objects from different api groups you would have to create a new kubeclient for that group.
Knowing what your target group is available through the apiVersion of the object:
```
apiVersion: batch/v1
kind: Job
```